File: two_sum/two_sum.py
class Solution():
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        # set() doesnt work because they ask us for the keys not the values
        diff = dict()

        for i, num in enumerate(nums):
            # print(i, ":",  num, abs(num-target))

            if abs(num-target) in diff:
                return [diff[abs(num-target)], i]
            else:
                diff[num] = i

        return []


print()
print(Solution().twoSum([3, 5, 8, 2, 13, 6], 7))
print(Solution().twoSum([2, 7, 11, 15], 9))
print(Solution().twoSum([], 9))

File: two_sum/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  //  set() doesnt work because they ask us for the keys not the values
  diff = {};

  for (let i = 0; i < nums.length; i++) {
    if (Math.abs(nums[i] - target) in diff) {
      return [diff[Math.abs(nums[i] - target)], i];
    } else {
      diff[nums[i]] = i;
    }
  }
  return [];
};

console.log(twoSum([3, 5, 8, 2, 13, 6], 7));
console.log(twoSum([2, 7, 11, 15], 9));
console.log(twoSum([], 9));
